### ISSUE TITLE
Persistent HTTP connections

### ISSUE THREAD

**Issue by grover** [zaphoyd/websocketpp#513]
The non-blocking HTTP handler (#425) introduced in 0.6.0 simplifies the implementation of HTTP handlers for most use cases. One part that websocketpp is missing though is support for HTTP/1.1 persistent connections, which enable a client to send multiple requests over the same socket. This is especially useful with TLS enabled as the TLS handshakes and TCP connection times can lead to a less-than ideal experience in the real world.

**Comment by zaphoyd** [comment]
The basic idea here is that instead of closing the TCP connection after an HTTP request when content-length bytes have been read, leave it open for some period of time and if any new bytes are read, treat them as additional HTTP requests?

**Comment by grover** [comment]
Yeah, the basic idea is to not close the socket after handling a single HTTP request, but go back into the read to wait for the next request. I have version that does that and seems to work, but is vulnerable to a slowloris type of attack. I will create a pull request with this soon and hope to get some input as to enhancements or changes I could make.

Regarding the slowloris, I'm currently thinking of enhancing this to prevent those types of attacks.
